### PR TITLE
Make it easier to identify incomplete assignments during GC

### PR DESCRIPTION
**Pull request by sargun**
Return specific error code on incomplete assignment as opposed to generic Unknown error

### CODE REVIEW

File: vpc/service/assign_addresses_v3.go [hunk 1]
```diff
@@ -298,6 +298,10 @@ WHERE assignment_id = $1
 				tracehelpers.SetStatus(err, span)
 				return nil, err
 			}
+		} else {
+			err = status.Errorf(codes.FailedPrecondition, "Assignment %s is not completed", assignmentID)
+			tracehelpers.SetStatus(err, span)
+			return nil, err
 		}
 		return nil, nil
```
Comment (solarkennedy) [inline vpc/service/assign_addresses_v3.go:306]: Do we still need this line here?

File: vpc/service/assign_addresses_v3.go [hunk 2]
```diff
@@ -298,6 +298,10 @@ WHERE assignment_id = $1
 				tracehelpers.SetStatus(err, span)
 				return nil, err
 			}
+		} else {
+			err = status.Errorf(codes.FailedPrecondition, "Assignment %s is not completed", assignmentID)
```
Comment (solarkennedy) [inline vpc/service/assign_addresses_v3.go:302]: This error indicates that we tried to assign an ip, but there was already some unfinished (incomplete) assignment in progress or something?

**Review by solarkennedy — APPROVED** [review]

File: vpc/service/assign_addresses_v3.go (hunk 2 above)
Comment (sargun) [reply vpc/service/assign_addresses_v3.go:302]: Right, and that the assignment is either in the process of being completed, or has failed to complete.

File: vpc/service/assign_addresses_v3.go (hunk 1 above)
Comment (sargun) [reply vpc/service/assign_addresses_v3.go:306]: yep, for the first case.

**Pull request merged**